be/src/vec/functions/array/function_array_contains_all.cpp

```diff
@@ -0,0 +1,189 @@
+        return Status::OK();
+    }
+
+private:
+    // the internal array type.
+    using Offset = ColumnArray::Offset64;
+    struct Array {
+        const ColumnPtr& data;     // data[i] is the i-th element of the array.
+        const NullMap& null_map;   // null_map[i] = true if data[i] is null.
+        const Offset start_offset; // the offset of the first element in the array.
+        const Offset end_offset;   // the offset of the last element in the array.
+
+        Array(const ColumnPtr& data_, const NullMap& null_map_, const Offset start_offset_,
+              const Offset end_offset_)
+                : data {data_},
+                  null_map {null_map_},
+                  start_offset {start_offset_},
+                  end_offset {end_offset_} {}
+    };
+
+    // construct an `Array` instance from a ColumnArray.
+    static Array make_array(const ColumnNullable* col_nullable,
+                            const ColumnArray::Offsets64& offsets, const size_t cur_row) {
+        const ColumnPtr& data = col_nullable->get_nested_column_ptr();
+        const NullMap& null_map = col_nullable->get_null_map_data();
+        const Offset start_offset = cur_row == 0 ? 0 : offsets[cur_row - 1] + 1;
+        const Offset end_offset = offsets[cur_row];
+
+        return Array(data, null_map, start_offset, end_offset);
+    }
+
+    // the semantics of this function is to check if the left array contains all of the right elements.
+    // it's important to note that the arrays are interpreted as sets, and hence the order of the elements
+    // and the number of occurrences of each element are not taken into account.
+    void _check_left_contains_all_right(const Array& left_array, const Array& right_array,
+                                        UInt8* result) {
+        static constexpr UInt8 CONTAINS_ALL = 1;
+        static constexpr UInt8 NOT_CONTAINS_ALL = 0;
+
+        // set the default result to NOT_CONTAINS_ALL.
+        *result = NOT_CONTAINS_ALL;
+
+        const bool left_has_nulls = !left_array.null_map.empty();
+        const bool right_has_nulls = !right_array.null_map.empty();
+
+        // the left array cannot contain all of the right elements if the right array has nulls while the left does not.
+        if (right_has_nulls && !left_has_nulls) {
+            return;
+        }
+
+        // for each element in the right array, check if it is contained in the left array.
+        // if any element is not contained, then the left array does not contain all of the right elements.
+        for (size_t j = right_array.start_offset; j <= right_array.end_offset; ++j) {
+            // skip null elements in the right array.
+            if (right_has_nulls && right_array.null_map[j]) {
+                continue;
+            }
+
+            // true if the current element is contained in the left array.
+            bool contained = false;
+            for (size_t i = left_array.start_offset; i <= left_array.end_offset; ++i) {
```

Comment:
here O(n) = j * i which is cost much more time .  can u just make O(n) = j+i ?
U can use function_array_map.h to help u get this time cost